### PR TITLE
Parallel build client/common/server/languages

### PR DESCRIPTION
**Pull request by mshima**
Move client/common/server/languages to same execution cycle, so they are queued in parallel.
Making configOptions synchronized between priorities.

Part of https://github.com/jhipster/generator-jhipster/issues/10528
https://github.com/jhipster/generator-jhipster/pull/10527 follow up

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -364,53 +364,9 @@ module.exports = class extends BaseGenerator {
...
-                });
-            },
-
-            askFori18n: prompts.askFori18n
-        };
-    }
-
-    get default() {
-        return {
+            askFori18n: prompts.askFori18n,
```
Comment (deepu105) [inline generators/app/index.js]: Doesn't this depend on value from the client?

**Comment by deepu105** [comment]
@mshima great work, these are the most complex parts of JHipster, I'm glad there are more people taking interest in these

File: generators/app/index.js (hunk 1 above)
Comment (mshima) [reply generators/app/index.js]: askFori18n shouldn't depend on value from client.

common/client/server modules depends on each other, but at default step, every module have configuration in place.

That the reason of https://github.com/mshima/generator-jhipster/commit/25c469bf7ee12a5538050ba8919a75a2c7823105

If you have doubt about yeoman workflow, take a look at the log from azure project step.
I've enabled yeoman debug.
And you can take a look at https://github.com/yeoman/generator/blob/2202fab7231707aeabbc2461d41e6bcafc743524/test/generators-compose-workflow.js

**Comment by mshima** [comment]
> * So any refactoring here needs to be carefully tested. I agree that the current code is quite complex as it grew over time. I also agree that this needs to be refactored, but needs to manually tested as our tests still don't cover all sort of use cases

That's the reason of https://github.com/jhipster/generator-jhipster/pull/10527, it should be safe for 6.x.

This PR should be for 7.x

**Comment by mraible** [comment]
Please fix conflicts in this pull request.

**Comment by mshima** [comment]
I will fix this PR once jhipster 7 cycle starts.

**Pull request closed without merging**